hq/app/logic/CredentialsReportDisplay.scala

```diff
@@ -56,40 +57,34 @@ object CredentialsReportDisplay {
   }
 
   def toCredentialReportDisplay(report: IAMCredentialsReport): CredentialReportDisplay = {
+    val humanUsers = report.entries.filterNot(_.rootUser).map {
+      case cred if cred.passwordEnabled.contains(true) =>
+        HumanUser(
+          cred.user,
+          cred.mfaActive,
+          accessKey1Details(cred),
+          accessKey2Details(cred),
+          humanReportStatus(cred),
+          dayDiff(lastActivityDate(cred)),
+          stack = cred.stack,
+          tags = cred.tags
+        )
+    }
 
-    report.entries.filterNot(_.rootUser).foldLeft(CredentialReportDisplay(report.generatedAt)) { (report, cred) =>
-      val machineUser =
-        if (!cred.passwordEnabled.getOrElse(false)) {
-          Some(MachineUser(
-            cred.user,
-            accessKey1Details(cred),
-            accessKey2Details(cred),
-            machineReportStatus(cred),
-            dayDiff(lastActivityDate(cred)),
-            stack = cred.stack,
-            tags = cred.tags
-          ))
-        } else None
-
-      val humanUser =
-        if (cred.passwordEnabled.getOrElse(false)) {
-          Some(HumanUser(
-            cred.user,
-            cred.mfaActive,
-            accessKey1Details(cred),
-            accessKey2Details(cred),
-            humanReportStatus(cred),
-            dayDiff(lastActivityDate(cred)),
-            stack = cred.stack,
-            tags = cred.tags
-          ))
-        } else None
-
-      report.copy(
-        machineUsers = report.machineUsers ++ machineUser,
-        humanUsers = report.humanUsers ++ humanUser
-      )
+    val machineUsers = report.entries.filterNot(_.rootUser).map {
+      case cred if !cred.passwordEnabled.getOrElse(false) =>
+        MachineUser(
+          cred.user,
+          accessKey1Details(cred),
+          accessKey2Details(cred),
+          machineReportStatus(cred),
+          dayDiff(lastActivityDate(cred)),
+          stack = cred.stack,
+          tags = cred.tags
+        )
     }
+
+    CredentialReportDisplay(report.generatedAt, machineUsers, humanUsers)
```

Comment:
the refactor of this method is perhaps the subjective one - but I think going from a `foldLeft` that conditionally appends an item to a list using `Option` and then copies the whole report is more complicated than 2 `filter`s and a single instantiation

Reply:
Agreed, I'm having some difficulty reading the original to make sure the behaviour is the same. This should be easy to test no? Run the application locally and see if the iam creds are listed in the same way as before in the UI?

Reply:
Yeah, it tests out fine locally!

Reply:
I ran this locally and the IAM creds are indeed listed in the same way as in PROD 👍🏽

Reply:
thanks! that clears it up